lib/forge.js

```diff
@@ -5,35 +5,64 @@ var router = require('./routes');
 var url = require('url');
 function Forge (opts) {
   var self = this;
-  self.initialize_tftpd();
-  self.initialize_httpd();
-  self.initialize_dhcpd(opts);
+  if(!opts)
+    opts = {};
+  
+  if(!opts.tftp)
+    opts.tftp = {};
+  self.initialize_tftpd(opts.tftp);
+  
+  if(!opts.http)
+    opts.http = {};
+  self.initialize_httpd(opts.http);
+  
+  if(!opts.tftp)
+    opts.dhcp = {};
+  self.initialize_dhcpd(opts.dhcp);
 }
 
 Forge.prototype.initialize_tftpd = function initialize_tftpd (opts) {
   var self = this;
   self.tftpd = tftpd.createServer({
-    host: '0.0.0.0',
-    port: 69,
-    root: './static/',
+    host: opts.host || '0.0.0.0',
+    port: opts.port || 69,
+    root: opts.root || './static/',
     denyPUT: true
   }).listen();
 };
 
 Forge.prototype.initialize_dhcpd = function initialize_dhcpd (opts) {
   var self = this;
+
+  if(!opts.store)
```

Comment:
Without a store, it doesn't make any sense to continue. The store will eventually have more methods associated with it that will deal with http and perhaps tftp, since a store may work against an API.